flist.c

```diff
@@ -231,10 +232,15 @@ static int readlink_stat(const char *path, STRUCT_STAT *stp, char *linkbuf)
 int link_stat(const char *path, STRUCT_STAT *stp, int follow_dirlinks)
 {
 #ifdef SUPPORT_LINKS
-	if (copy_links)
+	if (copy_links && update_links == 0)
```

Comment:
If they asked for --copy-links they should get --copy-links. Remember that it's only the sending side that sees this flag set, not the receiving side, so combining --copy-links with --update-links would mean that the sender doesn't send any symlinks, and the receiver could deal with any existing symlinks in the destination using the new option. (Thus, drop this change.)

Reply:
These two modes appear mutually exclusive from a user standpoint. Either every symlink is transformed into referent (`--copy-links`) or copied across if newer (`--update-links`). 

> so combining --copy-links with --update-links would mean that the sender doesn't send any symlink

From testing the PR'd code briefly now, when both modes are enabled, `--update-links` trumps `--copy-links`, by virtue of a subsequent condition at `if (update_links && S_ISLNK(stp->st_mode))`.

Added #274 to discuss how to handle symlink mode handling precedence.